src/main/java/org/folio/rest/impl/CalendarAPI.java

```diff
@@ -280,8 +256,136 @@ public void getCalendarPeriodsByServicePointIdPeriod(String servicePointId, bool
+    } else{
+      loanStartDateTime = ZonedDateTime.of(LocalDate.parse(startDate).atTime(LocalTime.now()), ZoneId.of("UTC"));
+      loanEndDateTime = loanStartDateTime.plusHours(amount);
+    }
+
+    Criteria critOpeningServicePointId = new Criteria().addField(SERVICE_POINT_ID).setJSONB(true).setOperation(Criteria.OP_EQUAL).setValue("'" + servicePointId + "'");
+    Criteria critOpeningByStartDateTime = new Criteria().addField(START_DATE).setJSONB(true).setOperation(Criteria.OP_LESS_THAN_EQ).setValue(loanStartDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME));
+    Criteria critOpeningByEndDateTime = new Criteria().addField(END_DATE).setJSONB(true).setOperation(Criteria.OP_GREATER_THAN_EQ).setValue(loanEndDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME));
+    Criterion criterionForOpeningHours = new Criterion();
+    criterionForOpeningHours.addCriterion(critOpeningServicePointId, Criteria.OP_AND);
+    criterionForOpeningHours.addCriterion(critOpeningByStartDateTime, Criteria.OP_AND, critOpeningByEndDateTime);
+
+    vertxContext.runOnContext(a ->
+      postgresClient.startTx(beginTx ->
+        postgresClient.get(beginTx, OPENINGS, Openings.class, criterionForOpeningHours, true, false, resultOfSelectOpenings -> {
+          if (resultOfSelectOpenings.succeeded()) {
+            addOpeningPeriodsToCollection(openingCollection, resultOfSelectOpenings);
+            if (CollectionUtils.isNotEmpty(openingCollection.getOpeningPeriods()) && openingCollection.getOpeningPeriods().size() == 1) {
+              String openingId = openingCollection.getOpeningPeriods().get(0).getId();
+              getOpening3DaysByOpeningIdFuture(asyncResultHandler, openingCollection, lang, openingId, loanEndDateTime, postgresClient, beginTx);
+            } else {
+              postgresClient.endTx(beginTx, done ->
+                asyncResultHandler.handle(Future.succeededFuture(GetCalendarPeriodsByServicePointIdPeriodByPeriodIdResponse.withPlainNotFound(messages.getMessage(lang, MessageConsts.ObjectDoesNotExist)))));
+            }
+          } else {
+            postgresClient.endTx(beginTx, done ->
+              asyncResultHandler.handle(Future.succeededFuture(GetCalendarPeriodsByServicePointIdPeriodByPeriodIdResponse.withPlainInternalServerError(messages.getMessage(lang, MessageConsts.InternalServerError)))));
+          }
+        })));
+  }
+
+  private void saveRegularHours(PostCalendarPeriodsRequestParams postCalendarPeriodsRequestParams, PostgresClient postgresClient, Handler<AsyncResult<Response>> asyncResultHandler, AsyncResult<Object> beginTx, AsyncResult<String> replyOfSavingOpenings) {
+    if (replyOfSavingOpenings.succeeded()) {
+      RegularHours regularHours = new RegularHours(replyOfSavingOpenings.result(), postCalendarPeriodsRequestParams.getOpeningsTable().getId(), postCalendarPeriodsRequestParams.getEntity().getOpeningDays());
+      postgresClient.save(REGULAR_HOURS, regularHours, replyOfSavingRegularHours -> {
```

Comment:
request to the database is not in a transaction. 
plz use save(AsyncResult<SQLConnection> sqlConnection, .......)